onedal/spmd/basic_statistics/incremental_basic_statistics.py

```diff
@@ -0,0 +1,78 @@
+# ==============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from daal4py.sklearn._utils import get_dtype
+
+from ..._device_offload import support_usm_ndarray
+from ...basic_statistics import (
+    IncrementalBasicStatistics as IncrementalBasicStatistics_nonSPMD,
+)
+from ...datatypes import _convert_to_supported, to_table
+from .._base import BaseEstimatorSPMD
+
+
+class IncrementalBasicStatistics(BaseEstimatorSPMD, IncrementalBasicStatistics_nonSPMD):
+    def _reset(self):
+        self._partial_result = super(
+            IncrementalBasicStatistics_nonSPMD, self
+        )._get_backend("basic_statistics", None, "partial_compute_result")
+
+    @support_usm_ndarray()
+    def partial_fit(self, X, weights=None, queue=None):
+        """
+        Computes partial data for basic statistics
+        from data batch X and saves it to `_partial_result`.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data batch, where `n_samples` is the number of samples
+            in the batch, and `n_features` is the number of features.
+
+        queue : dpctl.SyclQueue
+            If not None, use this queue for computations.
+
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        if not hasattr(self, "_queue"):
+            self._queue = queue
+        policy = super(IncrementalBasicStatistics_nonSPMD, self)._get_policy(queue, X)
+        X, weights = _convert_to_supported(policy, X, weights)
+
+        if not hasattr(self, "_onedal_params"):
+            dtype = get_dtype(X)
+            self._onedal_params = self._get_onedal_params(False, dtype=dtype)
+
+        X_table, weights_table = to_table(X, weights)
+        self._partial_result = super(
+            IncrementalBasicStatistics_nonSPMD, self
+        )._get_backend(
+            "basic_statistics",
+            None,
+            "partial_compute",
+            policy,
+            self._onedal_params,
+            self._partial_result,
+            X_table,
+            weights_table,
+        )
+
+    @support_usm_ndarray()
+    def finalize_fit(self, queue=None):
+        return super().finalize_fit(queue=queue)
```

Comment:
Doesn't inheritance handle this definition? Superfluous?

Reply:
it is done similar to all others algos in onedal/spmd and if you want to change that then it should be done in a separate pr I guess

Reply:
I believe its necessary because of the addition of support_usm_ndarray

Reply:
As per the documentation of (https://github.com/intel/scikit-learn-intelex/blob/main/onedal/_device_offload.py#L180): 
```    Handles USMArray input. Puts SYCLQueue from data to decorated function arguments.
    Converts output of decorated function to dpctl.tensor/dpnp.ndarray if input was of this type.

    Parameters
    ----------
    freefunc (bool) : Set to True if decorates free function.
    queue_param (bool) : Set to False if the decorated function has no `queue` parameter

    Notes
    -----
    Queue will not be changed if provided explicitly```

This method 1) only takes a queue as input 2) returns a copy of the estimator object.  Neither of which require the use of support_usm_ndarray. Therefore this code is not necessary.

Reply:
Good catch